### PR TITLE
Fixes an edge case in custom pagination link processing

### LINKED ISSUE #1886: Inconsistent Behavior of Links in Starlight | Frontmatter Prev and Next
### What version of `starlight` are you using?

0.22.4

### What version of `astro` are you using?

4.8.6

### What package manager are you using?

pnpm

### What operating system are you using?

Windows

### What browser are you using?

Brave

### Description

I've been experimenting with Starlight for a couple of days and have recently set up a project hosted on GitHub Pages. I've noticed that the behavior of the `Prev` and `Next` links is inconsistent and I'm unsure why this is happening.

### Details

The issue is that sometimes the links work as expected, and other times they don't. It seems that sometimes they need to be prefixed with the Astro config's `base` value, and other times they automatically prefix themselves.

To understand why the behavior changes, I created a fresh template install. In this template, I have three files in an auto-generated directory. Each file uses the exact same link. The link in `three.md` works, but the links in `one.md` and `two.md` do not.

In my actual documentation, I've tried prefixing the frontmatter links with the base value. Everything works with this prefixing, except for one file. I've also noticed odd differences where a link works in one test run and not in another. It seems whether the link is treated as relative or absolute, and whether it needs the base as a prefix, changes unpredictably.

### Steps to Reproduce

1. Create a fresh Starlight template.
2. Create a directory at `/src/content/docs/myfiles` within the new project.
3. Add three files (`one.md`, `two.md`, `three.md`) and add them to the sidebar config as auto-generated directory.
4. Use the same link in each file.

```js
---
title: one
sidebar:
    order: 1
next:
    label: example ref
    link: reference/example/
---

---
title: two
sidebar:
    order: 2
next:
    label: example ref
    link: reference/example/
---

---
title: three
sidebar:
    order: 3
next:
    label: example ref
    link: reference/example/
---
```

6. Observe that the link in `three.md` works, but the links in `one.md` and `two.md` do not.

### Expected Behavior

The links should consistently work across all files and should either all require manual prefixing with the base value or non of the links require the prefixing.

### Actual Behavior

The links work inconsistently. Sometimes they require manual prefixing with the base value, and other times they do not.

I believe this is not the intended behavior, hence I'm opening this issue. Any help or guidance would be greatly appreciated.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-rzkjm9?file=astro.config.mjs

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by delucis**
#### Description

- Closes #1886
- Fixes an edge case how custom prev/next links in frontmatter are handled
- In general a custom prev/next link was applied directly to an existing link object, which produced the correct behaviour. How in the edge case where a link object wasn’t already available (e.g. pagination is disabled, or the page is first/last, so has no prev/next link), we were using a `makeLink()` utility designed for sidebar items that did some additional processing. This rarely caused issues, but if a user had `base` set, this led to inconsistencies where the base would prepended to links **only** in these rare edge cases.
- This PR fixes this by ensuring links created for pagination purposes are never processed and always reflect 1:1 with the authored link in frontmatter
- We had an old test that expected the processing, which converted `x` to `/x`, but I think that was a wrong expectation, so I updated that old test to reflect the new behaviour.

### CODE REVIEW

File: packages/starlight/__tests__/basics/pagination-with-base.test.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		sidebar = getSidebar('/reference/frontmatter/', undefined);
+	});
+
+	test('pagination links are formatted correctly with no frontmatter', () => {
+		const links = getPrevNextLinks(sidebar, true, {});
+		expect(links.prev?.href).toMatchInlineSnapshot(`"/guides/components/"`);
+		expect(links.next?.href).toMatchInlineSnapshot(`undefined`);
+	});
+
+	test('pagination links are formatted correctly with custom links in frontmatter', () => {
```
Comment (HiDeoo) [inline packages/starlight/__tests__/basics/pagination-with-base.test.ts:35]: Tiny thought: should the test name not mention "formatted correctly" but more something like "used as authored" to emphasize we're not formatting the link in this case?

**Review by HiDeoo — APPROVED** [review]
Having a new function not involving `formatPath()` definitely seems like the proper solution to fix this issue. The updated test also makes sense to me.

Thanks for fixing this :raised_hands:

File: packages/starlight/__tests__/basics/pagination-with-base.test.ts (hunk 1 above)
Comment (delucis) [reply packages/starlight/__tests__/basics/pagination-with-base.test.ts]: Good point — updated and also switched away from snapshots to hardcoded values as I don’t think these should really change.
